### PR TITLE
Aik 3171 : Get context from WSGI to an Agent

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/agent/__init__.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    """
+    Our agent thread
+    """
+
+    def __init__(self, q):
+        logger.debug("Agent thread started")
+        while True:
+            while not q.empty():
+                self.process_data(q.get())
+        time.sleep(AGENT_SEC_INTERVAL)
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py:23]: this might still be resource intensive

File: aikido_firewall/agent/__init__.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+"""
+Aikido agent, this will create a new thread and listen for stuff sent by our sources and sinks
+"""
+
+import time
+import queue
+from threading import Thread
+from aikido_firewall.helpers.logging import logger
+
+AIKIDO_IPC_PORT = 49155
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py]: there is no ipc used right?

File: aikido_firewall/agent/__init__.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+AGENT_SEC_INTERVAL = 0.05
```
Comment (willem-delbare) [inline aikido_firewall/agent/__init__.py]: only required every 60 sec

Comment (bitterpanda63) [reply aikido_firewall/agent/__init__.py]: Done

File: aikido_firewall/agent/__init__.py (hunk 2 above)
Comment (bitterpanda63) [reply aikido_firewall/agent/__init__.py]: Nop, removed this useless constant

File: aikido_firewall/agent/__init__.py (hunk 1 above)
Comment (bitterpanda63) [reply aikido_firewall/agent/__init__.py:23]: interval changed now :0

File: aikido_firewall/context/__init__.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            ),
+        )
+
+    def set_as_current_context(self):
+        """
+        Set the current context
+        """
+        # pylint: disable=global-statement # We need this so that
+        # Sinks can access context
+        global current_context
```
Comment (willem-delbare) [inline aikido_firewall/context/__init__.py]: is dit dan threadsafe?

**Pull request merged**